pkg/auth/is_jwt_token_expired.go

```diff
@@ -0,0 +1,17 @@
+package auth
+
+import (
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
+// This method returns true if JWT token is expired, otherwise returns false
+func IsJWTTokenExpired(accessToken string) bool {
```

Comment:
This method returns false if the provided accessToken is invalid. If we are going this route I think this should be documented in the method's doc.

I am also wondering whether this is the way we want to approach it. We could decide to return a `bool, error` instead.
Regarding the `no Keyfunc was provided` error being returned by Parse, that's because the `Parse` method requires a `keyFunc`.
Then the next question is whether we want to ensure signature verification when parsing the token. Depending on the decision taken this should be at least documented if it is not evident. If Token verification is desired then maybe the `GetJWTFromSignedToken` from the AuthHelper data type could be used (although I haven't looked into details what it implies). If no token verification is desired then https://pkg.go.dev/github.com/golang-jwt/jwt/v4@v4.2.0#Parser.ParseUnverified can be used, although with extreme caution as verification is not being done.

An insight on `IsJWTTokenExpired` is that it performs two things:
1. Parsing a raw token
2. Check the expiration

An alternative would be accepting a jwt v4 token (which is already parsed) and just perform the expiration check.

Reply:
Yea, considering the specified use, this could lead to some future issues if misunderstood. Thanks for the clarification @akoserwal 

As this is called in:

Reply:
Yea, considering the specified use, this could lead to some future issues if misunderstood. Thanks for the clarification @akoserwal

As this is called in:
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/keycloak/client.go#L166
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/redhatsso/client.go#L97

Would I keep it separate in the client folder, or in keycloak/client.go as previously suggested?